core/runtime/common/executor.hpp

```diff
@@ -105,17 +107,18 @@ namespace kagome::runtime {
     outcome::result<Result> callAtGenesis(std::string_view name,
                                           Args &&...args) {
       OUTCOME_TRY(env_template, env_factory_->start());
-      OUTCOME_TRY(env, env_template->make());
+      OUTCOME_TRY(env, env_template->make(kNoCb));
       return callMediateInternal<Result>(
           *env, name, std::forward<Args>(args)...);
     }
 
     outcome::result<common::Buffer> callAtRaw(
```

Comment:
I really dislike that this codec is leaking through the codebase like this. Maybe we can have a separate implementation of TrieStorage, since it's a lightweight class, that would return batches that are able to deliver the trie proof, and explicitly state in StateApiImpl that we need to use this particular implementation of TreeStorage so that we can access the proof without changing the main TrieStorage interface?